src/lib/useQueryParams.ts

```diff
@@ -0,0 +1,110 @@
+import queryString, { ParseOptions } from 'query-string';
+import { useEffect } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import GradeLevel from '../types/GradeLevel';
+import sortOptions from './sortOptions';
+
+const options: ParseOptions = {
+  arrayFormat: 'bracket-separator',
+  arrayFormatSeparator: ',',
+  parseNumbers: true,
+};
+
+type SetQueryParamFn = (
+  key: 'minAmount' | 'maxAmount' | 'grades' | 'majors' | 'sortBy',
```

Comment:
I removed `qParams` in favor of this. As we move things to Typescript this'll prevent `setQueryParam` from being called with an unknown key.